src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,715 @@
+
+            # No need to construct for properties format
+            if format_ == 'properties' or separator is None or not separator:
+                exported_dict.update({key: kv.value})
+                continue
+
+            key_segments = key.split(separator)
+            if key_segments[0].isdigit():
+                __export_keyvalue(key_segments,
+                                  kv.value, exported_list, key)
+
+            else:
+                __export_keyvalue(key_segments,
+                                  kv.value, exported_dict, key)
+
+        if exported_dict and exported_list:
+            logger.error("Can not export to a valid file! Some keys have been dropped. %s", json.dumps(
+                exported_dict, indent=2, ensure_ascii=False))
+
+        return __compact_key_values(exported_dict if not exported_list else exported_list)
+    except Exception as exception:
+        raise CLIError("Fail to export key-values." + str(exception))
+
+
+def __export_features(retrieved_features, format_):
+    exported_dict = {}
+    client_filters = []
+
+    if format_ in ('yaml', 'properties'):
+        # We only support json feature flags for now
+        logger.warning("Exporting feature flags to a yaml or properties file is not supported yet. Ignoring all feature flags.")
+        return exported_dict
+
+    if format_ == 'json':
+        exported_dict["FeatureManagement"] = {}
+
+    try:
+        # retrieved_features is a list of FeatureFlag objects
+        for feature in retrieved_features:
+
+            # if feature state is on or off, it means there are no filters
+            if feature.state == "on":
+                feature_state = True
+
+            elif feature.state == "off":
+                feature_state = False
+
+            elif feature.state == "conditional":
+                feature_state = {"EnabledFor": []}
+                client_filters = feature.conditions["client_filters"]
+                # client_filters is a list of dictionaries, where all dictionaries have 2 keys - Name and Parameters
+                for filter_ in client_filters:
+                    feature_filter = {}
+                    feature_filter["Name"] = filter_.name
+                    if filter_.parameters:
+                        feature_filter["Parameters"] = filter_.parameters
+                    feature_state["EnabledFor"].append(feature_filter)
+
+            feature_entry = {feature.key: feature_state}
+
+            if format_ == 'json':
+                exported_dict["FeatureManagement"].update(feature_entry)
+
+        return __compact_key_values(exported_dict)
+
+    except Exception as exception:
+        raise CLIError("Failed to export feature flags. " + str(exception))
+
+
+def __convert_feature_dict_to_keyvalue_list(features_dict, format_):
+    # pylint: disable=too-many-nested-blocks
+    key_values = []
+    default_conditions = {'client_filters': []}
+    default_value = FeatureFlagValue(id_="")
+
```

Comment:
can you create a new FeatureFlagValue instance inside of the for loop in line 613? Otherwise, we have to be careful that every iteration, we will clean the last round default. The default_conditions does make sense as it is an actual default value.